### ISSUE TITLE
When trying to sweep a paper wallet, I/O error electrum..... (Url)  ECONNECTREFUSED appears 

### ISSUE THREAD

**Issue by Wikinaut** [bitcoin-wallet/bitcoin-wallet#394]
(The exact error message text might differ)

I don't know why the message appears, but after trying several times, the message went away and it now works.

I suggest you better catch this error and show a correct message to the user.

**Comment by Wikinaut** [comment]
@schildbach Mehr Details auf Anfrage.

**Comment by schildbach** [comment]
Yes, the process of sweeping paper wallets can be improved a lot. When Biteasy closed down, I quickly implemented an Electrum replacement but didn't improve on it. I'll summarize my suggestions:

- Add more Electrum servers to https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/assets/electrum-servers.txt (Testnet) and https://github.com/bitcoin-wallet/bitcoin-wallet/blob/prod/wallet/assets/electrum-servers.txt (Mainnet). Note currently for Mainnet I hand-picked two servers that got a valid SSL certificate (most servers are either just self-signed or not authenticated at all).

- Change the https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/src/de/schildbach/wallet/ui/send/RequestWalletBalanceTask.java so that multiple servers are queried in parallel and the results compared. I/O errors and parsing errors would make the whole server be ignored, at least for the ongoing query.

- Implement the Electrum server discovery, which sadly still uses IRC like Bitcoin used to do until half a decade ago. I'm hesitant with this, let's put it this way: our experiences with IRC Java libraries were not great.

**Comment by schildbach** [comment]
For, I added support for Electrum servers with self-signed certificates in c8f6d952cf678a719697f30719d3aa3e9a664df4.

**Comment by karloluiten** [comment]
Odd thing is, when I try to import/sweep a wallet in "Bitcoin Wallet" (not testnet) from F-Droid, it tries to connect to electrum servers on testnet [?]/

**Comment by schildbach** [comment]
In version 7.09, I implemented the first two points from my list above. So I'm closing this.

@karloluiten You're probably seeing one mainnet electrumx that just happens to have "testnet" in its domain name. If you think it's something else and you can reproduce this with 7.09, please don't hesitate to open a separate issue.
